src/extensions/vb/Microsoft.DotNet.UpgradeAssistant.Steps.VisualBasic/EnableMyDotSupportSubStep.cs

```diff
@@ -0,0 +1,106 @@
+    public class EnableMyDotSupportSubStep : UpgradeStep
+    {
+        private readonly VisualBasicProjectUpdaterStep _parent;
+
+        public override string Id => typeof(EnableMyDotSupportSubStep).FullName!;
+
+        public override string Title => "Update vbproj to support \"My.\" namespace";
+
+        public override string Description => "Modifies the vbproj file to address compilation errors.";
+
+        public EnableMyDotSupportSubStep(VisualBasicProjectUpdaterStep vbProjectUpdaterStep, ILogger logger)
+            : base(logger)
+        {
+            if (logger is null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            ParentStep = _parent = vbProjectUpdaterStep ?? throw new ArgumentNullException(nameof(vbProjectUpdaterStep));
+        }
+
+        protected override Task<bool> IsApplicableImplAsync(IUpgradeContext context, CancellationToken token)
+        {
+            // VB updates don't apply until a project is selected
+            if (context?.CurrentProject is null)
+            {
+                return Task.FromResult(false);
+            }
+
+            // applicable to Unit test projects and class libraries
+            return Task.FromResult(context.CurrentProject.OutputType == ProjectOutputType.Library);
+        }
+
+        protected override Task<UpgradeStepInitializeResult> InitializeImplAsync(IUpgradeContext context, CancellationToken token)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            return Task.FromResult(new UpgradeStepInitializeResult(UpgradeStepStatus.Incomplete, $"VB updater \"{Id}\" needs to be applied to {context.CurrentProject!.FileInfo}", BuildBreakRisk.Unknown));
+        }
+
+        protected override async Task<UpgradeStepApplyResult> ApplyImplAsync(IUpgradeContext context, CancellationToken token)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var file = context.CurrentProject!.GetFile();
+            if (string.IsNullOrWhiteSpace(file.GetPropertyValue("VBRuntime")))
+            {
+                // resolves errors
+                // 1>vbc : error BC30002: Type 'Global.Microsoft.VisualBasic.ApplicationServices.User' is not defined.
+                // 1>vbc : error BC30002: Type 'Global.Microsoft.VisualBasic.MyServices.Internal.ContextValue' is not defined.
+                // 1>vbc : error BC30002: Type 'Global.Microsoft.VisualBasic.ApplicationServices.User' is not defined.
+                // 1>vbc : error BC30002: Type 'Global.Microsoft.VisualBasic.Devices.Computer' is not defined.
+                // 1>vbc : error BC30002: Type 'Global.Microsoft.VisualBasic.ApplicationServices.ApplicationBase' is not defined.
+                file.SetPropertyValue("VBRuntime", "Embed");
```

Comment:
In the linked GitHub issue, devs mentioned `<MyType>`. Do we need to set that property also or is `<VBRuntime>` sufficient?

Reply:
We only need to set one of the two. Changing the &lt;MyType&gt; value can have side effects in other files so we're changing the &lt;VBRuntime&gt;

Reply:
That seems fine.